lib/transmuxer.js

```diff
@@ -1250,7 +1283,19 @@ Transmuxer = function(options) {
     self = this,
     videoTrack,
     audioTrack,
-    timelineStartDts,
+
+    // Store information about the start of the timeline in both DTS/
+    // PTS timescales. The start of a timeline is where in PTS/DTS time
+    // we were when the base media decode time was last (re)set to zero
+    timelineStartInfo = {},
+
+    // Signal that we are in the process of resetting the base media
+    // decode time.
+    // One effect of this state is that we trim audio frames that start
+    // earlier than the video track so that audio always has an initial
+    // base media decode time that is equal to or greater than the video
+    // track.
+    resettingBaseMediaDecodeTime = true,
```

Comment:
I have what may be an irrational dislike for boolean state variables. It may be because once you get started with them, it's so easy to add another without considering the additional "modes" you have created for an object. Are we only resetting the base decode when seeking or playing over a discontinuity? Could we simply discard and recreate the Transmuxer stack in that case?